index.py

```diff
@@ -226,69 +248,84 @@ def generate_positional_indexes_from_list(self, words, start_index):
                 last_position[word] = i
         return positions
 
-    def process_words(self, words):
+    def generate_token_list(self, doc_id, field_type, positional_indexes):
         """
-        Stems the already lowercase version of the word given and lowercases
-        Takes in the list of Strings and returns their stemmed version in a list
+        Generates entries from a list of positional index
         """
-        stemmer = nltk.stem.porter.PorterStemmer()
-        return [stemmer.stem(w.lower()) for w in words]
+        tokens_list = []
+        for term, positions in positional_indexes.items():
+            tokens_list.append([term, (doc_id, field_type, position)])  # [term, (doc_ID, Field, position)]
+
+        return tokens_list
 
     def calculate_doc_length(self):
         """
-        Sets and stores the length of each document for normalization
+        Sets and stores the length of each document for use during normalization
         """
-        self.doc_lengths = {}
         # Iterate through every term of the dictionary, getting its PostingList
-        # to iterate every posting and calculate its contribution to its document's vector length
-        # then, complete calculation for the vector's length
+        # We iterate every posting in the PostingList and calculate its contribution to its document's vector length
+        # This contribution is accumualted, and then square-rooted to find the vector's length, used for normalisation later on
+        self.doc_lengths = {}
         for _, posting_list in self.dictionary.items():
+
+            # This is done at term-level (aka for every term, aka using each PostingList)
+
+            # This stores doc_id:total_tf mappings
+            tf_overall = {}
+            # Accumulate the total_tf values
             for posting in posting_list.postings:
-                posting_weight = 1 + math.log(len(posting.positions), 10)
-                if posting.doc_id not in self.doc_lengths:
-                    self.doc_lengths[posting.doc_id] = posting_weight * posting_weight
+                tf_contribution = len(posting.positions) # tf contribution for current term from current zone/field
+                if (posting.doc_id) not in tf_overall:
+                    tf_overall[posting.doc_id] = tf_contribution
                 else:
-                    self.doc_lengths[posting.doc_id] += (posting_weight * posting_weight)
+                    tf_overall[posting.doc_id] += tf_contribution
+
+            # Since each term has a non-zero tf contribution to give a non-zero length contribution (due to lnc document weighting scheme)
+            # to the length of the document vector if the term appears in the document vector,
+            # we calculate this length contribution to the document length
+            for id, tf_value in D1.items():
```

Comment:
Cant seem to find D1's definituion, you mean tf_overall?

Reply:
Ahh yes, oversight! Thanks!